src/wp-includes/global-styles-and-settings.php

```diff
@@ -255,3 +255,70 @@ function ( $item ) {
 		}
 	}
 }
+
+/**
+ * Whether a theme or its parent have a theme.json file.
+ *
+ * @since 6.2.0
+ *
+ * @return bool
+ */
+function wp_theme_has_theme_json() {
```

Comment:
Should this logic live in a method in the `WP_Theme` class? See is_block_theme as a example. 

https://github.com/WordPress/wordpress-develop/blob/e27c5a38e371e1db85efeea239accf535227ed39/src/wp-includes/class-wp-theme.php#L1486-L1506

https://github.com/WordPress/wordpress-develop/blob/e50c65caf4999b11eb396ceaf6649b9818fab47f/src/wp-includes/theme.php#L4209-L4218

Reply:
> As noted in the PR https://github.com/WordPress/gutenberg/pull/45955. We need to know on EVERY request if the current AND parent theme support theme json. For that reason, this function is not fit for the requirements.

Different people have different views on this. @felixarntz mentioned it does not necessarily [requires modifying this function](https://github.com/WordPress/gutenberg/pull/45955#pullrequestreview-1220128114). I also shared the implementation can be different. There's agreement on querying themes other than the active theme is needed, though the implementation is not clear yet. Even if that PR was the way to go: it's a future modification that does not affect the API. We should aim to ship work in small pieces to make progress.

> Should this logic live in a method in the WP_Theme class? See is_block_theme as a example.

Note that there is a `wp_is_block_theme` as well, which is the main API for consumers, like this functions aims to be.

Whether we also have a `WP_Theme->is_block_theme` is something to talk about. This was brought up in the conversations we had in Gutenberg already. There was no clarity about value, so the decision was to ship the pieces that are ready and keep iterating on the parts that needed more thought.

Reply:
We didn't really discuss this as part of the gutenberg PR, as that is not a gutenberg change, it is a core change. Now we are in core, we need to talk about it again.

Reply:
I disagree @spacedmonkey. Smaller iterations of work can be committed into Core.

What do I mean exactly? For this particular PR's scope of work, it can be iteratively committed as separate initiatives such as:

* Initiative 1: Add functionality to provide a global way to identify if a theme supports `theme.json` is useful for Core.
* Initiative 2: Improve its performance, if needed.

Rather than blocking a backport, let the first get committed with a follow-up to improve its performance.

Reply:
Hmm that's interesting thought https://github.com/WordPress/wordpress-develop/pull/3556#discussion_r1050266726 @felixarntz. I agree. 

It's common for Core to have a global convenience function that wraps around a public method.

I agree:
* `WP_Theme::has_theme_json()` makes sense.
* and can be added in a separate PR and commit, ie for smaller scopes of work.

Reply:
> WP_Theme::has_theme_json() makes sense.

See https://github.com/WordPress/wordpress-develop/pull/3604/ where I did this

Reply:
@spacedmonkey The implementation from #3604 _uses_ the `WP_Theme` method in `wp_theme_has_theme_json()` function though, which is different from what @hellofromtonya and I were referring to further above though.

Normally I would say it makes sense and we don't want to duplicate code, but what I am outlining in https://github.com/WordPress/wordpress-develop/pull/3556#discussion_r1050266726 is a common core pattern already:
* For example, `get_stylesheet()` _does not_ call `WP_Theme::get_stylesheet()`.
* Instead, it implements the behavior in a simpler and less expensive way (without relying on `WP_Theme`).
* That pattern I think we should follow here as well.

We can create the `WP_Theme` method in a separate PR though, it doesn't have to be the same one.